src/basic/random-util.c

```diff
@@ -26,6 +30,41 @@
 #include "random-util.h"
 #include "time-util.h"
 
+
+int rdrand64(uint64_t *ret) {
+
+#ifdef __x86_64__
+        static int have_rdrand = -1;
+        unsigned char err;
+
+        if (have_rdrand < 0) {
+                uint32_t eax, ebx, ecx, edx;
+
+                /* Check if RDRAND is supported by the CPU */
+                if (__get_cpuid(1, &eax, &ebx, &ecx, &edx) == 0) {
+                        have_rdrand = false;
```

Comment:
nitpick: I'd use `have_rdrand = 0` since it's an int... (Yes, I know `false` is the same, but it's still a bit confusing I find.)

Reply:
hmm, we use "true" and "false" all over the place already for "bool-like" variables, and this one is one of those (i.e. a tri-state true/fals/dontknow value)